improc/simulator.py

```diff
@@ -453,6 +501,331 @@ def get_star_flux_values(self):
+        new_number = next((len(p) for p in pars if isinstance(p, (list, np.ndarray))), None)
+        if new_number is not None:
+            number = new_number
+
+        # check all list/array parameters are of consistent length
+        for p, i in enumerate(pars):
+            if isinstance(p, (list, np.ndarray)) and len(p) != number:
+                raise ValueError(
+                    f'Size mismatch between parameters: {len(p)} vs {number} (parameter {i+2}).')
+
+        if mean_x_pos is None:
+            mean_x_pos = rng.uniform(-0.01, 1.01, self.galaxy_number) * imsize[1]
+        elif isinstance(mean_x_pos, (int, float)):
+            mean_x_pos = np.full(number, mean_x_pos)
+        if mean_y_pos is None:
+            mean_y_pos = rng.uniform(-0.01, 1.01, self.galaxy_number) * imsize[0]
+        elif isinstance(mean_y_pos, (int, float)):
+            mean_y_pos = np.full(number, mean_y_pos)
+
+        self.galaxy_mean_x_pos = np.append(self.galaxy_mean_x_pos, mean_x_pos)
+        self.galaxy_mean_y_pos = np.append(self.galaxy_mean_y_pos, mean_y_pos)
+
+        if fluxes is None:
+            alpha_f = abs(self.galaxy_flux_power_law) - 1
+            fluxes = self.galaxy_min_flux / rng.power(alpha_f, number)
+        elif isinstance(fluxes, (int, float)):
+            fluxes = np.full(number, fluxes)
+        self.galaxy_mean_fluxes = np.append(self.galaxy_mean_fluxes, fluxes)
+
+        if widths is None:
+            alpha_w = abs(self.galaxy_flux_power_law) - 1
+            widths = self.galaxy_min_width / rng.power(alpha_w, number)
+        elif isinstance(widths, (int, float)):
+            widths = np.full(number, widths)
+        self.galaxy_width_values = np.append(self.galaxy_width_values, widths)
+
+        if exp_scale_ratios is None:
+            exp_scale_ratios = rng.uniform(0.5, 2, number) * widths
+        elif isinstance(exp_scale_ratios, (int, float)):
+            exp_scale_ratios = np.full(number, exp_scale_ratios)
+        self.galaxy_exp_scale_ratios = np.append(self.galaxy_exp_scale_ratios, exp_scale_ratios)
+
+        if sersic_scale_ratios is None:
+            sersic_scale_ratios = rng.uniform(0.5, 2, number) * widths
+        elif isinstance(sersic_scale_ratios, (int, float)):
+            sersic_scale_ratios = np.full(number, sersic_scale_ratios)
+        self.galaxy_sersic_scale_ratios = np.append(self.galaxy_sersic_scale_ratios, sersic_scale_ratios)
+
+        if sersic_flux_ratios is None:
+            sersic_flux_ratios = 10 ** rng.uniform(-2, -1, number)
+        elif isinstance(sersic_flux_ratios, (int, float)):
+            sersic_flux_ratios = np.full(number, sersic_flux_ratios)
+        self.galaxy_sersic_flux_ratios = np.append(self.galaxy_sersic_flux_ratios, sersic_flux_ratios)
+
+        if cos_is is None:
+            cos_is = rng.uniform(0.1, 1, number)
+        elif isinstance(cos_is, (int, float)):
```

Comment:
Worth verifying that it's between 0.1 and 1?

What will happen if you pass 0?  I believe you'll get an infinitely thin disk, since the model is just a 2-d sersic profile, and doesn't have any sort of disk thickness built in.  (As such, it's probably worth verifying that cos_is isn't too small.)